src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// write code here
+const employeeList = document.querySelector('ul');
+
+function sortList(listContainer) {
+  const listElements = Array.from(listContainer.children);
+
+  listElements.sort((a, b) => {
+    const salaryA = parseSalary(a.dataset.salary);
+    const salaryB = parseSalary(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+
+  listElements.forEach((item) => listContainer.appendChild(item));
+}
+
+function getEmployees(listContainer) {
+  return Array.from(listContainer.children).map((item) => ({
+    name: item.textContent.trim(),
+    position: item.dataset.position,
+    salary: parseSalary(item.dataset.salary),
+    age: Number(item.dataset.age),
```

Comment:
Similarly, ensure that the HTML elements have `data-position` and `data-age` attributes set correctly. If these attributes are missing, the `getEmployees` function will not retrieve the expected data.